File: Library1/SQLiteTest.fs
module AppRunningLogger.SQLiteTest

open SQLite.Net
open SQLite.Net.Attributes
open SQLite.Net.Interop
open SQLite.Net.Platform.Win32
open System
open System.Collections.Generic

type AppDefinition () =
    [<PrimaryKey;Unique>]
    member val Id : int64 = 0L with get, set
    [<Unique>]
    member val Path : string = "" with get, set
    interface IComparable<AppDefinition> with
        member this.CompareTo(o : AppDefinition) = this.Id.CompareTo(o.Id)

type AppRunningLog () =
    static let dtUnder = DateTime.MinValue
    [<PrimaryKey;AutoIncrement>]
    member val Id : int64 = 0L with get, set
    [<NotNull;Indexed>]
    member val AppId : int64 = 0L with get, set
    member val Active : bool = false with get, set
    member val Begin : DateTime = dtUnder with get, set
    member val End : DateTime = dtUnder with get, set

type DBConnection = SQLiteConnection

let newSQLiteConnection dbName =
    new SQLiteConnection(
            new SQLitePlatformWin32(),
            dbName,
            true
        )

let initialSettings = function
    | (conn : SQLiteConnection) ->
        conn.ExecuteScalar<string> "PRAGMA synchronous = NORMAL" |> ignore
        conn.ExecuteScalar<string> "PRAGMA journal_mode = WAL" |> ignore
        ()

/// <summary>
/// 
/// </summary>
let initMainDB () =
    let conn = newSQLiteConnection "main.db"
    initialSettings conn
    conn.CreateTable<AppDefinition>() |> ignore
    conn.CreateTable<AppRunningLog>() |> ignore
    conn

let getAppDefinition (conn : DBConnection) =
    conn.Table<AppDefinition>()

let addAppDefinition (conn : DBConnection) (appDefs : IEnumerable<AppDefinition>) =
    conn.InsertAll appDefs |> ignore

let getAppRunningLog (conn : DBConnection) minId =
    conn.Table<AppRunningLog>().Where(fun x -> x.Id >= minId)

File: ConsoleApplication1/SQLiteTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SQLite.Net;
using SQLite.Net.Attributes;
using SQLite.Net.Interop;
using SQLite.Net.Platform.Win32;

namespace AppRunningLogger
{
    class SQLiteTest
    {
        class AppDefinition : IComparable<AppDefinition>
        {
            [PrimaryKey, Unique]
            long Id { get; set; } = 0;
            [Unique]
            string Path { get; set; } = "";

            public int CompareTo(AppDefinition other)
            {
                return Id.CompareTo(other.Id);
            }
        }

        class AppRunningLog
        {
            [PrimaryKey, AutoIncrement]
            long Id { get; set; } = 0;
            [NotNull, Indexed]
            long AppId { get; set; } = 0;
            bool Active { get; set; } = false;
            DateTime Begin { get; set; } = DateTime.MinValue;
            DateTime End { get; set; } = DateTime.MinValue;
        }

        SQLiteConnection newSQLiteConnection(string dbName)
        {
            return new SQLiteConnection(new SQLitePlatformWin32(), dbName, true);
        }
    }
}
